relayer/spy_relayer/src/relayer/solana_utils.ts

```diff
@@ -0,0 +1,597 @@
+    wormhole,
+  } = wormholeAddresses;
+  const [wormholeMessage] = await deriveMessagePda(
+    tokenTransferWithPayloadSignedVaa,
+    wormhole,
+  );
+  const wormholeClaim = await getClaimAddressSolana(
+    tokenBridge.toBase58(),
+    tokenTransferWithPayloadSignedVaa,
+  );
+  const [swimPayloadMessage] = await getSwimPayloadMessagePda(
+    wormholeClaim,
+    propellerProgram.programId,
+  );
+
+  const propellerFeeVault: PublicKey = await Token.getAssociatedTokenAddress(
+    ASSOCIATED_TOKEN_PROGRAM_ID,
+    TOKEN_PROGRAM_ID,
+    swimUsdMint,
+    propeller,
+    true,
+  );
+  const propellerRedeemer = await getPropellerRedeemerPda(
+    propellerProgram.programId,
+  );
+
+  const propellerRedeemerEscrowAccount: PublicKey =
+    await Token.getAssociatedTokenAddress(ASSOCIATED_TOKEN_PROGRAM_ID, TOKEN_PROGRAM_ID, swimUsdMint, propellerRedeemer, true);
+
+  const marginalPricePoolInfo = await getMarginalPricePoolInfo(
+    propeller,
+    propellerProgram,
+    twoPoolProgram,
+  );
+  const propellerEngineFeeTracker = await getPropellerFeeTrackerAddr(
+    swimUsdMint,
+    payer.publicKey,
+    propellerProgram.programId,
+  );
+
+  const propellerEngineFeeTrackerData =
+    await propellerProgram.account.feeTracker.fetch(propellerEngineFeeTracker);
+
+  const parsedVaa = await parseVaaTyped(tokenTransferWithPayloadSignedVaa);
+  const parsedTransferWithSwimPayload = parseTransferWithArbPayload(Buffer.from(parsedVaa.payload));
+  const swimPayload = parseSwimPayload(
+    Buffer.from(parsedTransferWithSwimPayload.extraPayload),
+  );
+
+  const requestUnitsIx = web3.ComputeBudgetProgram.requestUnits({
+    // units: 420690,
+    units: 900000,
+    additionalFee: 0,
+  });
+
+  const propellerData = await propellerProgram.account.propeller.fetch(
+    propeller,
+  );
+  const aggregator = propellerData.aggregator;
+
+  const [foreignEndpoint] = await deriveEndpointPda(
+    parsedVaa.emitterChain,
+    parsedVaa.emitterAddress,
+    tokenBridge,
+  );
+
+  const completePubkeys = await propellerProgram.methods
+    .completeNativeWithPayload()
+    .accounts({
+      propeller,
+      payer: payer.publicKey,
+      tokenBridgeConfig,
+      message: wormholeMessage,
+      claim: wormholeClaim,
+      swimPayloadMessage: swimPayloadMessage,
+      endpoint: foreignEndpoint,
+      to: propellerRedeemerEscrowAccount,
+      redeemer: propellerRedeemer,
+      feeRecipient: propellerFeeVault,
+      custody: custody,
+      swimUsdMint: swimUsdMint,
+      custodySigner,
+      rent: web3.SYSVAR_RENT_PUBKEY,
+      systemProgram: web3.SystemProgram.programId,
+      wormhole: wormholeAddresses.wormhole,
+      tokenProgram: splToken.programId,
+      tokenBridge: wormholeAddresses.tokenBridge,
+    })
+    .pubkeys();
+
+  const completeNativeWithPayloadIxs = propellerProgram.methods
+    .propellerCompleteNativeWithPayload()
+    .accounts({
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
```

Comment:
What's the error here?

Reply:
it was some weird thing related to the anchor idl typedefs

Reply:
Can you be more specific?

Reply:
this is from a different line but the error should be similar
```
Type 'Partial  >' is not assignable to type 'Accounts{ name: "propeller"; isMut: false; isSigner: false; pda: { seeds: [{ kind: "const"; type: "string"; value: "propeller"; }, { kind: "account"; type: "publicKey"; account: "Propeller"; path: "propeller.swim_usd_mint"; }]; }; } | ... 18 more ... | { ...; }>'.
```